x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/calculations/utils.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ExpressionFunctionAST } from '@kbn/interpreter/common';
+import { IndexPattern, IndexPatternLayer } from '../../../types';
+import { ReferenceBasedIndexPatternColumn } from '../column_types';
+
+/**
+ * Checks whether the current layer includes a date histogram and returns an error otherwise
+ */
+export function checkForDateHistogram(layer: IndexPatternLayer) {
+  const buckets = layer.columnOrder.filter((colId) => layer.columns[colId].isBucketed);
+  const hasDateHistogram = buckets.some(
+    (colId) => layer.columns[colId].operationType === 'date_histogram'
+  );
+  if (hasDateHistogram) {
+    return undefined;
+  }
+  return [
+    i18n.translate('xpack.lens.indexPattern.calculations.dateHistogramErrorMessage', {
+      defaultMessage: 'Needs a date histogram to work',
```

Comment:
I think this error message would be improved by adding more context, like `Cumulative sum requires a date histogram to work. Choose a different function or add a date histogram.` This context might need to be passed in as an argument to the function